runtime/src/iree/tooling/trace_replay.c

```diff
@@ -532,8 +533,11 @@ static void iree_trace_replay_write_element(
     IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(UINT_16, uint16_t)
     IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(UINT_32, uint32_t)
     IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(UINT_64, uint64_t)
-    IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(FLOAT_32, float)
-    IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(FLOAT_64, double)
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_16:
+      *(uint16_t*)dst = iree_math_f32_to_f16((float)value);
+      break;
+      IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(FLOAT_32, float)
+      IREE_TRACE_REPLAY_WRITE_ELEMENT_CASE(FLOAT_64, double)
```

Comment:
misleading intent. you might need `// clang-format off` or maybe more this `case:` to just before the `default:`.